x-pack/plugins/apm/public/components/fleet_integration/apm_agents/agent_instructions_accordion.tsx

```diff
@@ -97,95 +106,143 @@ function TutorialConfigAgent({
 }
 
 interface Props {
+  policy: PackagePolicy;
   newPolicy: NewPackagePolicy;
+  onChange: PackagePolicyEditExtensionComponentProps['onChange'];
   agentName: AgentName;
   title: string;
   variantId: string;
   createAgentInstructions: CreateAgentInstructions;
+  AgentRuntimeAttachment?: (props: AgentRuntimeAttachmentProps) => ReactElement;
 }
 
 export function AgentInstructionsAccordion({
+  policy,
   newPolicy,
+  onChange,
   agentName,
   title,
   createAgentInstructions,
   variantId,
+  AgentRuntimeAttachment,
 }: Props) {
   const docLinks = useKibana().services.docLinks;
   const vars = newPolicy?.inputs?.[0]?.vars;
   const apmServerUrl = vars?.url.value;
   const secretToken = vars?.secret_token.value;
   const steps = createAgentInstructions(apmServerUrl, secretToken);
+  const stepsElements = steps.map(
+    (
+      { title: stepTitle, textPre, textPost, customComponentName, commands },
+      index
+    ) => {
+      const commandBlock = replaceTemplateStrings(
+        Array.isArray(commands) ? commands.join('\n') : commands || '',
+        docLinks
+      );
```

Comment:
Just wondering why we want to replace the newlines escape sequences (`\n`) here, and not below for:

```ts
markdown={replaceTemplateStrings(textPre, docLinks)}
```

It would make more sense to have a method that handled this


```suggestion
      const commandBlock = renderMustache({text: commands, docLinks, replaceNewline: true });
```